src/main/scala/device/MemEncrypt.scala

```diff
@@ -0,0 +1,1201 @@
+/***************************************************************************************
+* Copyright (c) 2024 Institute of Information Engineering, Chinese Academy of Sciences
+*
+* XiangShan is licensed under Mulan PSL v2.
+* You can use this software according to the terms and conditions of the Mulan PSL v2.
+* You may obtain a copy of Mulan PSL v2 at:
+*          http://license.coscl.org.cn/MulanPSL2
+*
+* THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND,
+* EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT,
+* MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
+*
+* See the Mulan PSL v2 for more details.
+***************************************************************************************/
+
+// See LICENSE.SiFive for license details.
```

Comment:
What's this? Is this code derived from rocket-chip?

Reply:
This code dose not derived from rocket-chip, just import some packages of rocket-chip. 
I didn't know if I should add a license for this, so I left it for now.
If it is not needed, I will delete it.

Reply:
import package does not need to add a license